pilot/test/mock/config.go

```diff
@@ -484,52 +485,34 @@ func CheckIstioConfigTypes(store model.ConfigStore, namespace string, t *testing
 
 // CheckCacheEvents validates operational invariants of a cache
 func CheckCacheEvents(store model.ConfigStore, cache model.ConfigStoreCache, namespace string, n int, t *testing.T) {
+	n64 := int64(n)
 	stop := make(chan struct{})
 	defer close(stop)
-	ach, dch := make(chan bool, n), make(chan bool, n)
-	defer close(ach)
-	defer close(dch)
-	sad, sdd := 0, 0
+	added, deleted := atomic.NewInt64(0), atomic.NewInt64(0)
```

Comment:
Don't make this change. go discourage shared memory or cond pattern: 
"Do not communicate by sharing memory; instead, share memory by communicating."
https://blog.golang.org/share-memory-by-communicating

Reply:
Yes, that is generally true in go code, and I wouldn't allow this kind shared memory into code paths that execute in production. However, this is a test, and clarity of code under test far outweighs that "rule". What's going on in the test is _far_ more obvious using these atomics compared to using channels. Look at the final `Eventually`, it's very explicit: eventually these two values will both be `n`, or the test fails. Compare that to the loop with channels/switch, where that very clear predicate is muddled by handling timeout and writes on the channel. '

Further, your code busy-waits: there's no delay or sleeping, it just compares added and deleted to n as fast as it can in a loop (or, occasionally, gets notified on a change on the other channels). Given our tests execute in parallel, that means you're occupying a core doing nothing while we could be executing another test in parallel.

Finally, which do you think is more maintainable in the long run? Less code that's more explicit about its meaning always wins in my book.

Reply:
for exactly the same reason go doesn't provide a min/max library. I prefer this not to be unified under one util.

Reply:
Golang does in fact provide a max/min library: [it's right in the standard library](https://golang.org/pkg/math/).

However, a function that can be written in four lines:
```go
func min(a, b MyNumType) MyNumType {
    if a < b {
        return a
    }
    return b
}
```
is a pretty far cry from a channel based setup. I'm pretty confident I can write min correctly every time, I'm not confident I can write eventual polling via channels correctly every time.

Reply:
I'm fine with the change since it's in test. I changed these to channels to remove flakiness issues and race issues. Please make sure it passes flaky test (-count 1000) and -racetest before submission.